src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncTests.cs

```diff
@@ -4830,5 +4830,47 @@ class C
                 Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "{ await task; }").WithArguments("System.Runtime.CompilerServices.IAsyncStateMachine", "SetStateMachine").WithLocation(62, 37));
         }
 
+        [Fact, WorkItem(16531, "https://github.com/dotnet/roslyn/issues/16531")]
+        public void ArityMismatch()
+        {
+            var source = @"
+using System;
+using System.Threading.Tasks;
+using System.Runtime.CompilerServices;
+
+public class Program
+{
+    public async MyAwesomeType<string> CustomTask() { await Task.Delay(1000); return string.Empty; }
+}
+
+[AsyncMethodBuilder(typeof(CustomAsyncTaskMethodBuilder<,>))]
+public struct MyAwesomeType<T>
+{
+    public T Result { get; set; }
+}
+
+public class CustomAsyncTaskMethodBuilder<T, V>
```

Comment:
I think that is covered by [AsyncTasklikeGenericBuilder](http://source.roslyn.io/#Roslyn.Compilers.CSharp.Emit.UnitTests/CodeGen/CodeGenAsyncTests.cs,1b5b198cd57d4b02). As far as I can tell, it contains all the permutations with arities 0 and 1. 
The scenario where the container type is generic and the nested type is generic (with arity 1) is covered by T_IIT/g1 as `G<int>.BG<int>` and T_OOT/g5 as `G<>.BG<>`.
I tried another permutation, `G<int>.BG<>` but that is bad for other reasons.